src/routes/addons.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * This program has been developed by students from the bachelor
+ * Computer Science at Utrecht University within the Software Project course.
+ *
+ * © Copyright Utrecht University
+ * (Department of Information and Computing Sciences)
+ */
+
+import { Request, Response } from "express";
+import { Context } from "../context";
+import { AddonCategory } from "prisma/prisma-client";
+
+interface GetAddonsRequest extends Request {
+  // use this to get typed access to request data (e.g. body, query, ...)
+  // see install.ts for an example
+
+  query: {
+    page?: string;
+    category?: string;
+  };
+}
+
+export const getAddonsRoute =
+  (ctx: Context) => async (req: GetAddonsRequest, res: Response) => {
+    // implement the route here
+    // - use ctx.prisma to access your database
+    // - use res.status and res.json to send a response
+
+    const pageSize = 20;
+
+    // Extract page and category from the request body
+    const { page, category } = req.query;
+    const pageNumber = page ? parseInt(page, 10) : 1;
+    const categoryEnum = category ? (category as AddonCategory) : undefined;
+
+    if (!page && !category) {
+      // Neither page nor category is provided
+      const addons = await ctx.prisma.addon.findMany();
+      res.status(200).json(addons);
+    } else if (page && !category) {
+      // Only page is provided
+      const addons = await ctx.prisma.addon.findMany({
+        skip: (pageNumber - 1) * pageSize,
+        take: pageSize
+      });
+      res.status(200).json(addons);
+    } else if (!page && category) {
+      // Only category is provided
+      const addons = await ctx.prisma.addon.findMany({
+        where: {
+          category: categoryEnum
+        }
+      });
+      res.status(200).json(addons);
+    } else if (page && category) {
+      // Both page and category are provided
+      const addons = await ctx.prisma.addon.findMany({
+        where: {
+          category: categoryEnum
+        },
+        skip: (pageNumber - 1) * pageSize,
+        take: pageSize
+      });
+      res.status(200).json(addons);
+    }
```

Comment:
https://github.com/PolarExpress/marketplace-backend/blob/1c2665c1b5700dd1f0aabe3bfcc2a1ca7877ac7e/src/routes/addons.ts#L36-L65

A lot of things are wrong with this code:
- first of all, page is always defined (because it defaults to 1) so there is no need to check if page is undefined.
- `!page` will be true if page is a falsy value, which includes `""`, `0`, `NaN`, `false`, `undefined` and `null`. This means valid data can fail this validity check, which causes hard-to-detect bugs.
- this if statement chain is not expandable at all. what if we had to do this with 5 optional parameters? we would need 32 if-blocks!!
- you have code duplication in a lot of these if blocks: the response status + json line is the same no matter what, and most of the query is the same as well.

Reply:
`page` doesn't actually default to 1 and it's defined as optional in validation, so it can also be absent. But the check for page isn't even necessary because `pageNumber` is defined for that purpose already.